engine/api/src/main/java/io/deephaven/engine/table/vectors/ColumnVectors.java

```diff
@@ -0,0 +1,489 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.vectors;
+
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.Table;
+import io.deephaven.vector.ByteVector;
+import io.deephaven.vector.CharVector;
+import io.deephaven.vector.DoubleVector;
+import io.deephaven.vector.FloatVector;
+import io.deephaven.vector.IntVector;
+import io.deephaven.vector.LongVector;
+import io.deephaven.vector.ObjectVector;
+import io.deephaven.vector.ShortVector;
+import io.deephaven.vector.Vector;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Utility methods for constructing {@link Vector Vectors} from the columns of a {@link Table}, enabling inefficient
```

Comment:
"inefficient random-access" and "efficient bulk access" could maybe use a bit more explanation and exposition.